.github/workflows/ci_tests.yml

```diff
@@ -0,0 +1,194 @@
+name: CI - Tests
+
+on:
+  pull_request:
+  push:
+    branches:
+      - 'master'
+      - 'push-action/**'  # Allow pushing to protected branches (using CasperWA/push-protected)
+
+jobs:
+
+  pre-commit:
+    runs-on: ubuntu-latest
+    # timeout-minutes: 2
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -U setuptools wheel
+
+        while IFS="" read -r line || [ -n "${line}" ]; do
+          if [[ "${line}" =~ ^pre-commit.*$ ]]; then
+            pre_commit="${line}"
+          fi
+        done < requirements_dev.txt
+
+        while IFS="" read -r line || [ -n "${line}" ]; do
+          if [[ "${line}" =~ ^invoke.*$ ]]; then
+            invoke="${line}"
+          fi
+        done < requirements_docs.txt
+
+        pip install ${pre_commit} ${invoke}
+
+    - name: Test with pre-commit
+      run: SKIP=pylint,pylint-strategies,pylint-tests pre-commit run --all-files
+
+  pylint-safety:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 2
+
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
+
+    - name: Install dependencies
+      run: |
+        python -m pip install -U pip
+        pip install -U setuptools wheel
+        pip install -U -r requirements.txt -r requirements_dev.txt -r requirements_docs.txt
+        pip install -e .
+        pip install safety
+
+    - name: Run pylint
+      run: pylint --rcfile=.pylintrc --ignore-paths=oteapi_dlite/strategies/,tests/ --extension-pkg-whitelist='pydantic' -- *.py oteapi_dlite
+
+    - name: Run pylint - oteapi/strategies
+      run: pylint --rcfile=.pylintrc --extension-pkg-whitelist='pydantic' --disable=R,C -- oteapi_dlite/strategies
+
+    - name: Run pylint - tests
+      run: pylint --rcfile=.pylintrc --extension-pkg-whitelist='pydantic' --disable=C0415,W0621 -- tests
+
+    # Ignore ID 44715 for now.
+    # See this NumPy issue for more information: https://github.com/numpy/numpy/issues/19038
+    - name: Run safety
+      run: pip freeze | safety check --stdin --ignore 44715
```

Comment:
I think we can remove this now - but test this before actually removing it.
Maybe keep it here for now and add another PR that just removes this to test it.

Reply:
Tested this - still doesn't work :(